include/circt/Dialect/RTL/RTLStructure.td

```diff
@@ -138,6 +138,102 @@ def RTLModuleExternOp : RTLOp<"module.extern",
   let parser = "return ::parse$cppClass(parser, result);";
 }
 
+def RTLGeneratorTypeOp : RTLOp<"generator.type", 
+      [Symbol, HasParent<"mlir::ModuleOp">]> {
+  let summary = "RTL Generator Type";
+  let description = [{
+    the "rtl.generator.type" operation declares a type of generated module.  The
+    type of a module is independent of the external method of producing it.  It
+    is assumed that for well known types, multiple external tools might exist
+    which can process it.  The type may be a specialization of another type 
+    (single-inheritance).  Generator types list attributes required by instances
+    and are inherited.
+
+    For example:
+    generator.type @MEMORY, "Simple-Memory", 1, "write_latency", "read_latency"
+    generator.type @FIRMEM (parent: @MEMORY), "FIRRTL-Memory", 1, "ports"
```

Comment:
Why do you have a version number, and what does it mean?

Reply:
My assumption is the the definition of a schema for some generator will change over time and generators will support a variety of schemas.  This could be signaled with names which included versioning sub-strings in an ad-hoc manor, or in a consistent general way.  For example, FIRRTL memory nodes will have some schema today, and when that schema changes tomorrow, it would be nice for generators which lower those nodes to be able to handle both flavors/versions.

This cannot be solved by embedding the version as an attribute in the required parameter list as that wouldn't support multiple concurrent versions in the same file.